Signing plugin not applied

In 0.14.2 the following worked:

```
apply plugin: 'org.jetbrains.dokka'
apply plugin: 'com.vanniktech.maven.publish'

mavenPublish {
  releaseSigningEnabled = !getGpgKey().isEmpty()
}

def getGpgKey() {
  return findProperty("signingKey") ?: ""
}

signing {
  def signingKey = getGpgKey()
  if(!signingKey.isEmpty()) {
    useInMemoryPgpKeys(signingKey, "")
  }
}
```

After updating to 0.15.0 I get the following error:

```
Could not find method signing()
```

If I add:

```
apply plugin: 'signing'
```

the build works. This happens regardless of whether `releaseSigningEnabled` is `true` or `false`.

I also get the following error if I don't apply the signing plugin manually:

```
Invalid MavenPublish Configuration. Unable to find variant to publish named release. Try setting the 'androidVariantToPublish' property in the mavenPublish extension object to something that matches the variant that ought to be published.
```

This happens regardless of whether I manually set `androidVariantToPublish` or if I use the default.

I tried using `signingInMemoryKey` and now it fails when configuring the `CloseAndReleaseRepositoryTask`.

Here's what my gradle config looks like:

```
apply plugin: 'org.jetbrains.dokka'
apply plugin: 'com.vanniktech.maven.publish'

mavenPublish {
  sonatypeHost = "S01"
}
```

Here are the env vars I have set up:

```
ORG_GRADLE_PROJECT_mavenCentralUsername
ORG_GRADLE_PROJECT_mavenCentralPassword
ORG_GRADLE_PROJECT_mavenCentralRepositoryUsername
ORG_GRADLE_PROJECT_mavenCentralRepositoryPassword
ORG_GRADLE_PROJECT_signingKey
ORG_GRADLE_PROJECT_signingInMemoryKey
```

And here's the error:

```
> Task :debug-db-base:build

> Task :debug-db-base:dokkaHtml
Initializing plugins
Dokka is performing: documentation for debug-db-base
Validity check
Creating documentation models
Exiting Generation: Nothing to document


> Task :debug-db-base:dokkaJavadocJar
> Task :debug-db-base:generateMetadataFileForMavenPublication
> Task :debug-db-base:generatePomFileForMavenPublication
> Task :debug-db-base:signMavenPublication FAILED

57 actionable tasks: 56 executed, 1 up-to-date
FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':debug-db-base:signMavenPublication'.
> Cannot perform signing task ':debug-db-base:signMavenPublication' because it has no configured signatory

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 24s
```

--- a/src/main/kotlin/com/vanniktech/maven/publish/legacy/BaseSetup.kt
+++ b/src/main/kotlin/com/vanniktech/maven/publish/legacy/BaseSetup.kt
@@ -14,6 +14,7 @@ import org.gradle.api.JavaVersion
 import org.gradle.api.Project
 import org.gradle.api.tasks.javadoc.Javadoc
 import org.gradle.external.javadoc.StandardJavadocDocletOptions
+import org.gradle.plugins.signing.SigningPlugin
 import org.jetbrains.dokka.gradle.DokkaTask
 
 internal fun Project.configureMavenCentral() {
@@ -23,6 +24,10 @@ internal fun Project.configureMavenCentral() {
 }
 
 internal fun Project.configureSigning() {
+  // Apply signing immediately. It is also applied by `signAllPublications` but the afterEvaluate means
+  // that it's APIs are not available for consumers without also using afterEvaluate.
+  plugins.apply(SigningPlugin::class.java)
+
   afterEvaluate {
     if (legacyExtension.releaseSigningEnabled) {
       baseExtension.signAllPublications()

